rfc.md

```diff
@@ -0,0 +1,215 @@
+# biologica 2.0 RFC
+
+### Motivation 1
+
+To have a more maintainable library with type definitions
+
+### Motivation 2
+
+I have always hated needing to transform organisms back and forth between the
+BioLogica.Organism class objects and plain-object formats. Frequently in Geni*
+we need to turn objects into BioLogica.Organisms before we can use them, or,
+worse, check whether we've already been passed an Organism or a plain object.
+
+For this reason it would be nice to define an Organism so that it was simply a
+typed object, which could be serialized, read, and used without any further
+transformations.
+
+As a result of this, it seems that the library should consist only of pure
+functions which take Organisms (or other typed objects) as inputs (e.g.
+`createGametes(Organism)` instead of `Organism.createGametes()`). A worry is that
+this may result in some awkward syntax.
+
+## Organism definition and creation
+
+### Organism
+```
+  {
+    alleles: "a-a b-b ...",
+    sex: "female"
+    // optional
+    phenotype: {
+      trait: characteristic
+    },
+    // optional
+    secondXAlleles: "c d ..."
+  }
+```
+
+* alleles: string  
+   Fully-defined allele string. Two organisms with the same allele string will always
+   have the same phenotype. **Or should this be called `genotype`**?
+* sex: string  
+* phenotype: { [s: string]: string; }  
+   It's useful to be able to read the phenotype of an organism easily, e.g.
+   `createOrganism(...).phenotype.color` instead of
+   `org = createOrganism(...); getCharacteristic(org, 'wings')`, so it seems to make
+   sense for this to be a part of any organism generated by the library. However, it
+   is redundant information, so it probably doesn't need to be passed in when using
+   an organism.
+* secondXAlleles: string  
+   This formalizes the pattern from geni* that allows us to switch sexes without changing the
+   phenotype. **Maybe this is unneeded?** We could presumably also have a `changeSex` function
+   that explictly picks second-x-alleles that don't change the phenotype. Do we need this for
+   anything else?
+
+### Creation
+```
+  organism({
+    /** optional. One would pass in fully-specified alleles when re-generating
+      phenotypes */
+    alleles: "a-a b-b ..."
+    /** optional. Explicitly specifying the authored syntax */
+    authoredAlleles: "a-a b- -c d-[D1, D2] e-^[E1] XY"
+    /** optional */
+    sex: "male"
+  }): Organism
+
+  breed(org: Organism, org: Organism, crossover=true): Organism
```

Comment:
Slight preference for `male` and `female`...`mother` and `father` are a little too human for me

Reply:
Hard to come up with terms that both allow flexibility for different species but also have a generally-agreed upon ordering so that we have a standard for which comes first.

I'm inclined towards `mother` and `father` because in something like a plant, the plant itself may be both male and female, but one plant would be performing the "mother" role and one plant the "father" role (even if it's the same plant in both roles).

(...this discussion may be the definition of bikeshedding, but it's more fun discussing this than doing real work!)